src/etc/inc/shaper.inc

```diff
@@ -5272,11 +5272,20 @@ function filter_generate_dummynet_rules() {
 		if (!is_module_loaded("dummynet.ko")) {
 			mwexec("/sbin/kldload dummynet");
 		}
-		set_sysctl(array(
+		$sysctls = (array(
 				"net.inet.ip.dummynet.io_fast" => "1",
 				"net.inet.ip.dummynet.hash_size" => "256",
 				"net.inet.ip.dummynet.pipe_slot_limit" => $max_qlimit
 		));
+		init_config_arr(array('sysctl', 'item'));
+		if (!empty($config['sysctl']['item'])) {
```

Comment:
Not that it needs changed now, but for future reference, this test is also unnecessary. It's harmless to run foreach on an empty array, it is effectively a no-op since the body of the statement never executes.